pyproject.toml

```diff
@@ -53,7 +53,7 @@ dependencies = [
     "huggingface_hub==0.20.3",
     "torch>=2.0",
     "GitPython==3.1.31", # for logging
-    "datasets>=2.14.0",
+    "datasets>=2.14.0,<2.18.0",
```

Comment:
Why do we set it strictly less than 2.18 if this PR make modifications to be compatible with it ?

Reply:
I thought the problem was fixed with `trust_remote_code=True` but it was not it. However, since this constraint will be needed in future versions of dataset, let's keep it too.

Atm, there is a bug in 2.18 which seemed like it's linked to `fsspec`'s version.
@albertvillanova is investigating, but in the meantime let's upper bound this.